components/push/src/communications.rs

```diff
@@ -259,36 +261,58 @@ impl Connection for ConnectHttp {
     ///
     /// A `None` channel_id indicates that we are unsubscribing
     /// from all channels
-    fn unsubscribe(&mut self, channel_id: Option<&str>) -> error::Result<bool> {
+    fn unsubscribe(&mut self, channel_id: &str) -> error::Result<bool> {
```

Comment:
I don't want to derail this too much, but the `bool` return values smell here given `false` is never returned. Do you think this is worth cleaning up? (ditto with the new function below). Maybe the original intention was to inspect the response for a "not an error, but there was no subscription" type result?

Reply:
No, I think this is a great cleanup!! Thank you - I'll add it to the list on #4372 , I messed around with it, it'll be a breaking change (at least for the `unsubscribeAll` but I think it'll be worth the follow up